ui/src/components/WMWorkerTypesTable/index.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from 'react';
+import { arrayOf, string } from 'prop-types';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+import Typography from '@material-ui/core/Typography/';
+import LinkIcon from 'mdi-react/LinkIcon';
+import memoize from 'fast-memoize';
+import { camelCase } from 'change-case';
+import { isEmpty, map, pipe, sort as rSort } from 'ramda';
+import { WMWorkerTypeSummary } from '../../utils/prop-types';
+import DataTable from '../DataTable';
+import sort from '../../utils/sort';
+import Link from '../../utils/Link';
+import DateDistance from '../DateDistance';
+
+export default class WMWorkerTypesTable extends Component {
+  static propTypes = {
+    workerTypes: arrayOf(WMWorkerTypeSummary).isRequired,
+    searchTerm: string,
+    path: string,
+  };
+
+  state = {
+    sortBy: null,
+    sortDirection: null,
+  };
+
+  sortWorkerTypes = memoize(
+    (workerTypes, sortBy, sortDirection, searchTerm) => {
+      const sortByProperty = camelCase(sortBy);
+      const filteredWorkerTypes = searchTerm
+        ? workerTypes.filter(({ workerType }) =>
+            workerType.includes(searchTerm)
+          )
+        : workerTypes;
+
+      return isEmpty(filteredWorkerTypes)
+        ? filteredWorkerTypes
+        : [...filteredWorkerTypes].sort((a, b) => {
+            const firstElement =
+              sortDirection === 'desc' ? b[sortByProperty] : a[sortByProperty];
+            const secondElement =
+              sortDirection === 'desc' ? a[sortByProperty] : b[sortByProperty];
+
+            return sort(firstElement, secondElement);
+          });
+    },
+    {
+      serializer: ([workerTypes, sortBy, sortDirection, searchTerm]) => {
+        const ids = pipe(
+          rSort((a, b) => sort(a.workerType, b.workerType)),
+          map(({ workerType }) => workerType)
+        )(workerTypes);
```

Comment:
This will redefine the function every time the serializer function is triggered. It's okay to put this functional function at the top of the file. It won't be exported.